cpp/src/arrow/render_api.cpp

```diff
@@ -54,40 +57,51 @@ get_heatmap(std::shared_ptr<arrow::Array> arr_x, std::shared_ptr<arrow::Array> a
 
     std::pair<uint8_t* ,int64_t> output;
     switch(c_type) {
+        case arrow::Type::INT8 : {
+            auto input_c_int8 = (int8_t *) arr_c->data()->GetValues<uint8_t>(1);
+            return out_pic(heatmap<int8_t >(input_x, input_y, input_c_int8, x_length));
+        }
+        case arrow::Type::INT16 : {
+            auto input_c_int16 = (int16_t *) arr_c->data()->GetValues<uint8_t>(1);
+            return out_pic(heatmap<int16_t >(input_x, input_y, input_c_int16, x_length));
+        }
+        case arrow::Type::INT32 : {
+            auto input_c_int32 = (int32_t *) arr_c->data()->GetValues<uint8_t>(1);
+            return out_pic(heatmap<int32_t >(input_x, input_y, input_c_int32, x_length));
+        }
+        case arrow::Type::INT64 : {
+            auto input_c_int64 = (int64_t *) arr_c->data()->GetValues<uint8_t>(1);
+            return out_pic(output = heatmap<int64_t >(input_x, input_y, input_c_int64, x_length));
+        }
+        case arrow::Type::UINT8 : {
+            auto input_c_uint8 = (uint8_t *) arr_c->data()->GetValues<uint8_t>(1);
+            return out_pic(heatmap<uint8_t >(input_x, input_y, input_c_uint8, x_length));
+        }
+        case arrow::Type::UINT16 : {
+            auto input_c_uint16 = (uint16_t *) arr_c->data()->GetValues<uint8_t>(1);
+            return out_pic(heatmap<uint16_t >(input_x, input_y, input_c_uint16, x_length));
+        }
+        case arrow::Type::UINT32 : {
+            auto input_c_uint32 = (uint32_t *) arr_c->data()->GetValues<uint8_t>(1);
+            return out_pic(heatmap<uint32_t >(input_x, input_y, input_c_uint32, x_length));
+        }
+        case arrow::Type::UINT64 : {
+            auto input_c_uint64 = (uint64_t *) arr_c->data()->GetValues<uint8_t>(1);
+            return out_pic(heatmap<uint64_t >(input_x, input_y, input_c_uint64, x_length));
+        }
         case arrow::Type::FLOAT : {
             auto input_c_float = (float *) arr_c->data()->GetValues<uint8_t>(1);
-            output = heatmap<float>(input_x, input_y, input_c_float, x_length);
-            break;
+            return out_pic(heatmap<float>(input_x, input_y, input_c_float, x_length));
         }
         case arrow::Type::DOUBLE : {
             auto input_c_double = (double *) arr_c->data()->GetValues<uint8_t>(1);
-            output = heatmap<double>(input_x, input_y, input_c_double, x_length);
-            break;
-        }
-        case arrow::Type::UINT32 : {
-            auto input_c_uint32 = (uint32_t *) arr_c->data()->GetValues<uint8_t>(1);
-            output = heatmap<uint32_t >(input_x, input_y, input_c_uint32, x_length);
-            break;
+            return out_pic(heatmap<double>(input_x, input_y, input_c_double, x_length));
         }
+
         default:
             std::cout << "type error!";
```

Comment:
use log, not cout

Reply:
Log related code has not been written，I add now.